source/merge-deep.d.ts

```diff
@@ -0,0 +1,136 @@
+interface MergeDeepOptions {
+  strict?: boolean;
+}
+
+type UnknownRecord = Record<string, unknown>;
+
+type Unwrap<Type> = Type extends UnknownRecord
+  ? {[Key in keyof Type]: Type[Key]}
+  : Type;
+
+type MergeValue<Destination, Source, Key, Options> =
+  Key extends keyof Destination
+    ? Key extends keyof Source
+      ? Destination[Key] extends UnknownRecord
+        ? MergeDeep<Destination[Key], Source[Key], Options>
+        : Source[Key]
+      : Destination[Key]
+    : Key extends keyof Source
+      ? Source[Key]
+      : never;
+
+type Keyof<Destination, Source> = keyof Destination | keyof Source;
+
+type MergeArrayValue<Destination, Source, Options> =
+  Destination extends UnknownArray
+    ? Source extends UnknownArray
+      ? ArrayMerge<Destination, Source, Options>
+      : Source
+    : Source;
+
+type GetValue<
+  Destination,
+  Source,
+  Key extends Keyof<Destination, Source>,
+  Options,
+> = Key extends keyof Source
+  ? Source[Key] extends UnknownRecord
+    ? MergeValue<Destination, Source, Key, Options>
+    : Source[Key] extends UnknownArray
+			? Key extends keyof Destination
+				? MergeArrayValue<Destination[Key], Source[Key], Options>
+				: Source[Key]
+			: Source[Key]
+  : Key extends keyof Destination
+  ? Destination[Key]
+  : never;
+
+type ExtractType<TType, TUnion> = {
```

Comment:
Move it to https://github.com/sindresorhus/type-fest/blob/main/source/internal.d.ts and add doc comment.

Reply:
Replaced with `ConditionalExcept` which already exists...